i2c/imx6ull/libimx6ull-i2c.c

```diff
@@ -0,0 +1,361 @@
+ *  - register readout could be written by generating RESTART event instead of STOP / START (beneficial in multi-master mode)
+ */
+
+
+static void platform_init(void)
+{
+	platformctl_t pctl;
+	unsigned int i;
+
+	/* enable clock */
+	pctl.action = pctl_set;
+	pctl.type = pctl_devclock;
+	pctl.devclock.dev = clocks[i2c.dev_no - 1];
+	pctl.devclock.state = 0b11;
+
+	platformctl(&pctl);
+
+	/* set MUXes */
+	pctl.type = pctl_iomux;
+	pctl.iomux.sion = 1; /* from RM: SION bit needs to be enabled for i2c pads */
+
+	for (i = 0; i < 2; ++i) {
+		pctl.iomux.mux = i2c_pctl_mux[i2c.dev_no - 1][i].pctl;
+		pctl.iomux.mode = i2c_pctl_mux[i2c.dev_no - 1][i].val;
+		platformctl(&pctl);
+	}
+
+	/* set IOsel */
+	pctl.type = pctl_ioisel;
+
+	for (i = 0; i < 2; ++i) {
+		pctl.ioisel.isel = i2c_pctl_isel[i2c.dev_no - 1][i].pctl;
+		pctl.ioisel.daisy = i2c_pctl_isel[i2c.dev_no - 1][i].val;
+		platformctl(&pctl);
+	}
+
+	/* PAD configuration taken from linux Device Tree, not needed for our current targets, leaving commented-out
+	 * as it might need to be enabled for targets with missing pullup resistors (and written in generic way) */
+#if 0
+	pctl.type = pctl_iopad;
+	pctl.iopad.pad = pctl_pad_gpio1_02;
+	pctl.iopad.hys = 1;
+	pctl.iopad.pus = 0b10; /* 100kOhm pull-up */
+	pctl.iopad.pue = 0;    /* pull/keeper enabled */
+	pctl.iopad.pke = 1;
+	pctl.iopad.ode = 1;      /* open drain enable */
+	pctl.iopad.speed = 0b10; /* speed - 100 MHz */
+	pctl.iopad.dse = 0x4;
+	pctl.iopad.sre = 0;
+
+	platformctl(&pctl);
+
+	pctl.iopad.pad = pctl_pad_gpio1_03;
+	platformctl(&pctl);
+#endif
+}
+
+
+static int is_bus_busy(void)
+{
+	return *(i2c.base + i2sr) & (1 << 5);
+}
+
+
+static int wait_bus_busy(int for_busy)
+{
+	int timeout = 10; /* x 10 us = 100us */
+
+	while (!!is_bus_busy() != !!for_busy) {
+		usleep(10);
+		if (--timeout <= 0)
+			return -ETIMEDOUT;
+	}
+
+	return 0;
+}
+
+
+static int i2c_intr(unsigned int intr, void *data)
+{
+	/* clear interrupt */
+	*(i2c.base + i2sr) &= ~(1 << 1);
+
+	return i2c.cond;
+}
+
+
+static int i2c_trx_complete(void)
+{
+	const time_t timeout_us = 100; /* 100 us */
+	int ret = 0;
+
+	mutexLock(i2c.lock);
+
+	/* wait for transfer complete interrupt - with timeout */
+	while (!(*(i2c.base + i2sr) & (1 << 7))) {
+		if ((ret = condWait(i2c.cond, i2c.lock, timeout_us)) == -ETIME) {
+			ret = -ETIMEDOUT;
+			break;
+		}
+	}
+
+	/* clear interrupt second time - sticky conditionals */
+	*(i2c.base + i2sr) &= ~(1 << 1);
+
+	mutexUnlock(i2c.lock);
+	return ret;
+}
+
+
+static int i2c_acked(void)
+{
+	if (*(i2c.base + i2sr) & (1 << 0))
+		return -EIO;
+
+	return 0;
+}
+
+
+/* writes 1 byte and waits for completion / ack */
+static int write_byte(uint8_t byte)
+{
+	int ret = 0;
+
+	*(i2c.base + i2sr) = 0; /* clear previous interrupts */
+	*(i2c.base + i2dr) = byte;
+
+	/* wait for byte to be sent */
+	ret = i2c_trx_complete();
+	if (ret < 0)
+		return ret;
+
+	ret = i2c_acked();
+	return ret;
```

Comment:
```suggestion
	return  i2c_acked();
```

Reply:
will do, remnant of it being a part of other function